src/platform/plugins/shared/dashboard/public/dashboard_renderer/dashboard_renderer.tsx

```diff
@@ -49,27 +50,21 @@ export function DashboardRenderer({
   onApiAvailable,
 }: DashboardRendererProps) {
   const dashboardViewport = useRef(null);
-  const dashboardContainerRef = useRef<HTMLElement | null>(null);
   const [dashboardApi, setDashboardApi] = useState<DashboardApi | undefined>();
   const [dashboardInternalApi, setDashboardInternalApi] = useState<
     DashboardInternalApi | undefined
   >();
   const [error, setError] = useState<Error | undefined>();
 
+  const dashboardContainerRef = useStateFromPublishingSubject(
+    dashboardInternalApi?.dashboardContainerRef$ || new BehaviorSubject(null)
+  );
```

Comment:
sorry - I was wrong but didn't delete my original comment in time :)  see the updated comment here: https://github.com/elastic/kibana/pull/233226#discussion_r2349684690